#Robot Frames (CHECK URDF)
base_link: "pelvis"
lfoot: "leftFoot"
rfoot: "rightFoot"
modelname: "/home/master/talos_walk_ws/src/serow/share/urdf/valkyrie_sim.urdf"
#modelname: "/home/master/ros_ws/src/serow/share/urdf/valkyrie_sim_gazebo_sync.urdf"

useLegOdom: true
#ROS Topic Names
odom_topic: "/ihmc_ros/valkyrie/output/visual_odom" #only if usePoseUpdate is false and useLegOdom is false
imu_topic: "/pelvisMiddleImu/imu"
joint_state_topic: "/ihmc_ros/valkyrie/output/joint_states"
lfoot_force_torque_topic: "/ihmc_ros/valkyrie/output/foot_force_sensor/left"
rfoot_force_torque_topic: "/ihmc_ros/valkyrie/output/foot_force_sensor/right"


#Ground-Truth #Not Mandatory - For comparison only
ground_truth: true
ground_truth_odom_topic: "/valkyrie/ground_truth/odom"
ground_truth_com_topic: "/valkyrie/ground_truth/CoM"
is_in_ds_topic: "/ihmc_ros/valkyrie/output/capturability/is_in_double_support"
T_B_GT: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0 ,0, 1]


#To publish relative data
debug_mode: true

#TF from IMU to Body
T_B_A: [1, 0, 0, 0, 0, -1, 0, 0, 0, 0 ,-1 ,-0.108, 0, 0 ,0, 1]
T_B_G: [1, 0, 0, 0, 0, -1, 0, 0, 0, 0 ,-1 ,-0.108, 0, 0 ,0, 1]
#TF from F/T to Left Foot
T_FT_LL: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0 ,0, 1]
#TF from F/T to Right Foot
T_FT_RL: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0 ,0, 1]
#TF from External Pose to Body only if useLegOdom is false
#T_B_P: [0, 1, 0, 0, -1, 0, 0, 0, 0, 0 ,1 ,0, 0, 0, 0, 1]
T_B_P: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 ,1 ,0, 0, 0, 0 , 1]

#ROS Topic Frequencies in Hz
imu_topic_freq: 500
joint_topic_freq: 500
fsr_topic_freq: 500  

##Schmitt Trigger - Contact Classifier
LegUpThres: 250 #Schmitt Trigger High Threshold in Newtons (N)
LegLowThres: 125 #Schmitt Trigger Low Threshold in Newtons (N)
LosingContact: 80
StrikingContact: 1200
VelocityThres: 0.07
medianWindow: 13
useGEM: false
#Steady State Joint Kalman Filter (KF)
computeJointVelocity: False #True if not included in the joint_state msg
joint_cutoff_freq: 20
joint_noise_density: 0.25


#Mahony Filter for attitude
useMahony: true
Mahony_Kp: 0.02
Mahony_Ki: 0.0

#Madgwick Filter for attitude if useMahony is false
Madgwick_gain: 0.012

#Leg Odometry Coefficients
Tau0: 1.0
Tau1: 1.0


#Rigid body Extended Kalman Filter (EKF)
#To estimate:
# 3-D Body Position/Velocity
# 3-D IMU-Biases

#Process Noise STDs
accelerometer_bias_random_walk:  0.5 #m/s^2/sqrt(s)
gyroscope_bias_random_walk: 4.1884e-2 # rad/s/sqrt(s)
accelerometer_noise_density: 1.7 #m/s^2  #Continuous Time
gyroscope_noise_density: 0.2 #rad/s  #Continuous Time
useInIMUEKF: false
contact_random_walk: 0.2
#Measuremets
leg_odom_position_noise_density: 5.0e-03
leg_odom_orientation_noise_density: 5.0e-03

#Odometry Measurement Noise std:
odom_position_noise_density_x: 1.0e-2
odom_position_noise_density_y: 1.0e-2
odom_position_noise_density_z: 1.0e-2

odom_orientation_noise_density: 1.0e-2

velocity_noise_density_x: 1.5 #0.007
velocity_noise_density_y: 1.5 #0.009
velocity_noise_density_z: 1.5 #0.007

#IMU Initial Biases - Biases are substracted!
bias_ax: 0.0
bias_ay: 0.0
bias_az: 9.7997
bias_gx: -1.60298e-05
bias_gy: -1.24745e-05
bias_gz: -2.60968e-05
calibrateIMUbiases: true #otherwise specify accurate initial values for bias_a, bias_g
maxImuCalibrationCycles: 200
#Filter Params
gravity: 9.80 #gravity constant (m/s^2)


# CoM Extended Kalman Filter (EKF)
# To estimate:
# CoM 3-D Position/Velocity
# 3-D External Forces on the CoM
estimateCoM: true   #Use the CoM EKF

#---------------------------------------------#
#Specify the rest only if estimateCoM is true!!
#---------------------------------------------#

#Process Noise
com_position_random_walk: 1.0e-5 #in m
com_velocity_random_walk: 5.0e-4 #in m/s
external_force_random_walk: 2.5   #in Newtons
#Measurment Noise
com_position_noise_density: 1.0e-05  #CoM position in m
com_acceleration_noise_density: 0.09 #CoM Acceleration in m/s^2
#Initial errors in External Forces
bias_fx: -23.5  #in Newtons
bias_fy: -24.0
bias_fz: -55.0
#Gyro LPF
useGyroLPF: false
#Cutt-off Frequencies
gyro_cut_off_freq: 10.0  #Only if useGyroLPF is true
#Gyro Moving Average Filter
maWindow: 15 #buffer size, the larger the smoother and delayed the signal, only if useGyroLPF is false
#Filter Params
mass: 127.324  #robot mass
Ixx : 0.11866378  #torso inertia around x
Iyy : 0.097963425  #torso inetria around y
Izz : 0.083854638   #torso inertia around z
